src/mo_frontend/typing.ml

```diff
@@ -2544,6 +2569,42 @@ and infer_obj env s exp_opt dec_fields at : T.typ =
   end;
   t
 
+and validate_parenthetical env typ_opt = function
+  | None -> ()
+  | Some par ->
+     let env = { env with async = C.NullCap } in
+     begin match typ_opt with
+     | Some fun_ty when T.is_func fun_ty ->
+       let s, _, _, _, ts2 = T.as_func fun_ty in
+       begin match ts2 with
+       | _ when T.is_shared_sort s -> ()
+       | [cod] when T.is_async cod -> ()
+       | _ -> warn env par.at "M0210" "unexpected parenthetical note on a non-send call"
+       end
+     | _ -> ()
+     end;
+     let [@warning "-8"] par_infer env { it = ObjE (bases, fields); _ } =
+       let checked = T.[ { lab = "cycles"; typ = nat; src = empty_src}
+                       ; { lab = "timeout"; typ = nat32; src = empty_src}
+                       ] in
+       infer_check_bases_fields env checked par.at bases fields in
+     let attrs = infer_exp_wrapper par_infer T.as_immut env par in
+     let [@warning "-8"] T.Object, attrs_flds = T.as_obj attrs in
+     if attrs_flds = [] then warn env par.at "M0211" "redundant empty parenthetical note";
+     let check_lab { T.lab; typ; _ } =
+       let check want =
+         if not (sub env par.at typ want)
+         then local_error env par.at "M0127"(*FIXME*) "parenthetical field %s is declared with type%a\ninstead of expected type%a" lab
+                display_typ typ
+                display_typ want in
+       match lab with
+       | "cycles" -> check T.nat
+       | "timeout" -> check T.nat32
+       | _ -> () in
+     List.iter check_lab attrs_flds;
```

Comment:
This should probably go into `infer_check_bases_fields`, as it currently won't check bases...

Reply:
also, use `checked` above

Reply:
Done: 0c07b52f8